src/transformers/models/whisper/generation_whisper.py

```diff
@@ -745,6 +719,46 @@ def generate(
         if return_segments:
             return {"sequences": sequences, "segments": final_segments}
 
+        if is_shortform:
+            # add eos token:
+            if generation_config.max_new_tokens is None and generation_config.max_length is None:
+                # fmt: off
+                sequences = torch.cat(
+                    [
+                        sequences,
+                        torch.full((sequences.shape[0], 1,), generation_config.eos_token_id).to(sequences.device),
+                    ],
+                    dim=-1,
+                )
```

Comment:
here you should decalre `somthing =torch.full((sequences.shape[0], 1,), generation_config.eos_token_id).to(sequences.device),` then do sequences = torch.cat